regex.c

```diff
@@ -69,6 +96,14 @@ bool match_regex(char *input)
 		}
 	}
 
+	// If a regex filter matched, we additionally compare the domain
+	// against all known whitelisted domains to possibly prevent blocking
+	// of a specific domain. The logic herein is:
+	// If matched, then compare against whitelist
+	// If in whitelist, negate matched so this function returns: not-to-be-blocked
+	if(matched)
+		matched = !in_whitelist(input);
```

Comment:
Should this get moved to the dnsmasq hook code? It pollutes the purpose of the method a bit. Instead of just matching against the regex, it also handles some blocking logic.